### ISSUE TITLE
Handle new 'x-github-client://openRepo/' - `Open in Desktop` broken for GitHub.com

### ISSUE THREAD

**Issue by winkelsdorf** [ForkIssues/Tracker#503]
It looks like changes to github.com break #43. GitHub now uses the `x-github-client:` scheme instead of `github-mac:` with it's `Open in Desktop` links.

Can anybody confirm this? I knew `x-github-client` was previously tied to beta versions of GitHub Desktop but it looks like this is the new default protocol scheme associated. See https://github.com/desktop/desktop/issues/1657#issuecomment-330624478.

**Comment by darrix** [comment]
I think this is also broken in Windows 10 as well.  IE (!) will open the link with GitHub Desktop but Edge, Chrome, and Firefox all fail to use GD, taking me instead to the Github Desktop download page.  This is with version 1.6.2.  I've uninstalled GD, erased the install folder, reinstalled, and verified that the registry paths are correct, but nothing seems to correct the problem.

**Comment by NSBrianWard** [comment]
Yes, I'm having this problem too! I even went in to `~/Library/Preferences/com.apple.LaunchServices/com.apple.launchservices.secure.plist` and tried pointing anything that looked like it might be git-related (`sourcetree`, `x-github-client`, `github-mac`) to fork and it didn't help. Went so far as to changing the plist in the app bundle to try to get Fork to accept `x-github-client` links but that just seemed to cause a crash on codesigning.

**Comment by DanPristupov** [comment]
I just checked, for me it's still `github-mac://openRepo/https://github.com/ForkIssues/Tracker` on GitHub. 🤔

P.S. On Windows there's no such link at all.

**Comment by NSBrianWard** [comment]
I'm not sure how GitHub is deciding how to emit those links. I wish I could just make it emit `github-mac` links. I did previously have SourceTree installed. Perhaps something is still registered somewhere? Sourcetree does appear to use those `x-github-client` links fwiw.

**Comment by NSBrianWard** [comment]
On another Mac, it looks like GitHub Desktop responds to both `x-github-client` and `github-mac`, but GitHub is emitting those same `x-github-client` links.

**Comment by BenTilbrook** [comment]
GitHub seems to decide which URL based on your GitHub account.

Some time ago, I went through the process of installing GitHub Desktop. I didn't like it so I removed it and used SourceTree and eventually Fork as my handler. At this time, `github-mac` URLs were emitted from GitHub.

A week or so ago, I (regretfully) revisited GitHub Desktop and went through the same process, and now GitHub emits `x-github-client` URLs.

So, in short, +1! Please add `x-github-client` support to Fork!

**Comment by NoelJacob** [comment]
> GitHub seems to decide which URL based on your GitHub account.
> 
> Some time ago, I went through the process of installing GitHub Desktop. I didn't like it so I removed it and used SourceTree and eventually Fork as my handler. At this time, `github-mac` URLs were emitted from GitHub.
> 
> A week or so ago, I (regretfully) revisited GitHub Desktop and went through the same process, and now GitHub emits `x-github-client` URLs.
> 
> So, in short, +1! Please add `x-github-client` support to Fork!

Please take this into account

**Comment by NSBrianWard** [comment]
Any chance this will get fixed? I really wanna just use Fork, but I still need to keep Sourcetree around to handle easily cloning branches. It's ridiculously annoying.

**Comment by ISS2718** [comment]
Hey every one, 
I run this code and I managed to use the "Open in Desktop", but in Opera not worked Chrome and Edge only (at least in my tests).

---

```
Windows Registry Editor Version 5.00

[-HKEY_CLASSES_ROOT\github-windows]

[HKEY_CLASSES_ROOT\github-windows]
"URL Protocol"=""
@="URL:github-windows"

[HKEY_CLASSES_ROOT\github-windows\shell\open\command]
@="\"C:\\Users\\USERNAME\\AppData\\Local\\GitHubDesktop\\app-2.5.5\\GitHubDesktop.exe\" --protocol-launcher \"%1\""

[-HKEY_CLASSES_ROOT\x-github-client]

[HKEY_CLASSES_ROOT\x-github-client]
"URL Protocol"=""
@="URL:x-github-client"

[HKEY_CLASSES_ROOT\x-github-client\shell\open\command]
@="\"C:\\Users\USERNAME\\AppData\\Local\\GitHubDesktop\\app-2.5.5\\GitHubDesktop.exe\" --protocol-launcher \"%1\""

[HKEY_LOCAL_MACHINE\SOFTWARE\GitHubDesktop]

[HKEY_LOCAL_MACHINE\SOFTWARE\GitHubDesktop\Capabilities]

[HKEY_LOCAL_MACHINE\SOFTWARE\GitHubDesktop\Capabilities\UrlAssociations]
"github-windows"="github-windows"
"x-github-client"="x-github-client"

[HKEY_LOCAL_MACHINE\SOFTWARE\RegisteredApplications]
"GitHubDesktop"="Software\\GitHubDesktop\\Capabilities"
```
---

I found this solution [here](https://www.thetopsites.net/article/52747051.shtml)

**Comment by yvettep321** [comment]
I'M having the same problem

**Comment by NSBrianWard** [comment]
> Hey every one, I run this code and I managed to use the "Open in Desktop", but in Opera not worked Chrome and Edge only (at least in my tests).
> 
> ```
> Windows Registry Editor Version 5.00
> 
> [-HKEY_CLASSES_ROOT\github-windows]
> 
> [HKEY_CLASSES_ROOT\github-windows]
> "URL Protocol"=""
> @="URL:github-windows"
> 
> [HKEY_CLASSES_ROOT\github-windows\shell\open\command]
> @="\"C:\\Users\\USERNAME\\AppData\\Local\\GitHubDesktop\\app-2.5.5\\GitHubDesktop.exe\" --protocol-launcher \"%1\""
> 
> [-HKEY_CLASSES_ROOT\x-github-client]
> 
> [HKEY_CLASSES_ROOT\x-github-client]
> "URL Protocol"=""
> @="URL:x-github-client"
> 
> [HKEY_CLASSES_ROOT\x-github-client\shell\open\command]
> @="\"C:\\Users\USERNAME\\AppData\\Local\\GitHubDesktop\\app-2.5.5\\GitHubDesktop.exe\" --protocol-launcher \"%1\""
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\GitHubDesktop]
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\GitHubDesktop\Capabilities]
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\GitHubDesktop\Capabilities\UrlAssociations]
> "github-windows"="github-windows"
> "x-github-client"="x-github-client"
> 
> [HKEY_LOCAL_MACHINE\SOFTWARE\RegisteredApplications]
> "GitHubDesktop"="Software\\GitHubDesktop\\Capabilities"
> ```
> 
> I found this solution [here](https://www.thetopsites.net/article/52747051.shtml)

I tried doing something similar in macOS to force it but was unsuccessful. Would be nice to know if a fix is at least in the queue for this.

**Comment by artk42** [comment]
Hope to see any updates on that 🙏

**Comment by yvettep321** [comment]
It doesn't work

**Comment by byaruhaf** [comment]
+1! Please add x-github-client support to Fork!

**Comment by KnifeFed** [comment]
How come this is ignored? It means the "Default GitHub Tool" in Settings doesn't do anything, which becomes very confusing if you don't know about this issue.

**Comment by DanPristupov** [comment]
I think GitHub no longer uses`x-github-client` url scheme, does it?

**Comment by KnifeFed** [comment]
Yes, they do. I just tested it and clicking the "Checkout with GitHub Desktop" button just took me to https://desktop.github.com. After installing the GitHub Desktop app and reloading the page, it's now a `x-github-client://openRepo/<repo url>` link. That link remains after having uninstalled the app. This is on macOS Ventura.

**Comment by NSBrianWard** [comment]
> Yes, they do. I just tested it and clicking the "Checkout with GitHub Desktop" button just took me to https://desktop.github.com. After installing the GitHub Desktop app and reloading the page, it's now a `x-github-client://openRepo/<repo url>` link. That link remains after having uninstalled the app. This is on macOS Ventura.

This has been my experience as well. It changes after you install the GitHub Desktop app and there's no apparent way to change it back. Running on latest Sonoma 14.4.
